Create general start page, Closes #200 (#220)
LiskHQ/lisk-docs: start.md

# Lisk Documentation

Welcome to the Lisk documentation!

> Beside the **Lisk Protocol**, you find all you need to **setup, run and configure** the different parts of the **Lisk ecosystem**.

Reviewer: ```suggestion
Here you will find all you need to **setup, run and configure**  a Lisk Node and join the network. 
We also describe the **Lisk Protocol**  and **Lisk products**.
```

> ## The Lisk Protocol

Reviewer: Move to the bottom.

The [Lisk Protocol](../lisk-protocol/introduction.md) is a formal set of rules of the Lisk blockchain.

It consists of rules about the P2P communication, block and transaction formats, the consensus algorithm, and of rules how signatures and IDs are computed and verified.

The protocol defines these rules without focus on concrete implementation, and it is therefore possible to implement it in any compatible programming language.

> Currently, the official implementations of the Lisk protocol are [Lisk Core](../lisk-core/introduction.md) and [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md), which are written in Javascript and Typescript.

Reviewer: ```suggestion
Currently, the official implementations of the Lisk protocol are [Lisk Core](../lisk-core/introduction.md) and [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md), which are written in Typescript and [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md) written in Javascipt.
```

> ## The Lisk ecosystem

Reviewer: ```suggestion
## Lisk products
```

> The Lisk Ecosystem consists of a set of modular software products that provide developers with all the tools they need to develop their own custom blockchain applications.

Reviewer: ```suggestion
```

> It also consists of software that is needed to maintain and monitor the Lisk Main network.

Reviewer: ```suggestion
```

Tool                                                         | Description
------------------------------------------------------------ | --------------------------------------------------

> [Lisk SDK](../lisk-sdk/introduction.md) > [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md) | Modular set of building blocks with specific blockchain related functionality.

Reviewer: ```suggestion
[Lisk SDK](../lisk-sdk/introduction.md) |  Set of modular software products that provide developers with all the tools they need to develop their own custom blockchain applications.
[Lisk SDK](../lisk-sdk/introduction.md) > [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md) | Modular set of building blocks with specific, blockchain-related, functionality.
```

> [Lisk SDK](../lisk-sdk/introduction.md) > [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md)   | JS library for interacting with the network.

Reviewer: ```suggestion
[Lisk SDK](../lisk-sdk/introduction.md) > [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md)   | Set of libraries implementing Lisk protocol, contains all of the cryptography tools you may need to implement Lisk network interactions into your application.
```

[Lisk SDK](../lisk-sdk/introduction.md) > [Lisk Commander](../lisk-sdk/lisk-commander/introduction.md) | CLI-tool to interact with the network.

> [Lisk Hub](../lisk-hub/introduction.md)                      | UI for users to interact with the network, e.g. to manage their accounts, or to send or receive tokens.

Reviewer: ```suggestion
[Lisk Hub](../lisk-hub/introduction.md)                      | Lisk wallet and UI tool for users to interact with the network, e.g. to manage their accounts, send or receive tokens or vote for delegates.
```

[Lisk Core](../lisk-core/introduction.md)                    | The software needed to run a node on the Lisk Mainchain.
[Lisk Service](../lisk-service/introduction.md)              | Extended API interface for Lisk nodes and the extended network.

## Getting started

### Maintain a node

You want to become a node operator?

Great!

There are several good reasons for it:

> - Have your **private entry point** to [communicate](../lisk-core/api) with the network

Reviewer: ```suggestion
- Have your **private entry point** to [communicate](../lisk-core/api) with the network, it's important if you are an **exchange** implementing LSK tokens.
```

- **Full control** to [configure](../lisk-core/configuration.md) your node to your specific needs
- Create your own **snapshots** of the blockchain
- **Forge** new blocks (if you are an active delegate)

For the following user groups, it is recommended to set up an own node:

- Exchanges and other services that rely on a stable API interface to the network.
- Registered Delegates, who would like forge actively.
- Users who do not trust different sources and want to be in full control.

To become a node operator, you need to setup [Lisk Core](../lisk-core/introduction.md) on a server and connect it to the desired network.

### Build a blockchain application

#### What is a blockchain application?

> Blockchain applications are applications that communicate with a blockchain network in the background.

Reviewer: ```suggestion
Blockchain applications communicate within a network to agree upon which data is valid and should be stored and which should be rejected and bad actors potentially punished. The set of blockchain rules is already implemented by Lisk modules and libraries. By re-using them and implementing well-designed Lisk SDK Framework it's easy to compose new case-specific applications. Explore more great content of [Blockchain Business](https://lisk.io/academy/blockchain-business) in [Lisk Academy](https://lisk.io/academy).
```

> This blockchain network can be used as decentralized database layer to create, read, update or delete data within the system.

Reviewer: ```suggestion
```

> A **pure blockchain application** has no central data storage at all, and all information that is needed to perform its actions is actually stored within the blockchain itself.

Reviewer: ```suggestion
```

> A **hybrid blockchain application** may have certain data stored inside of a blockchain, but might additionally use centralized data storage solutions in certain areas.

Reviewer: ```suggestion
```

> #### And why do I need it?

Reviewer: Remove for now.

The Lisk SDK consists of the following tools:

- [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md)
- [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md)
- [Lisk Commander](../lisk-sdk/lisk-commander/introduction.md)

The Lisk Framework consists of a selection of various Modules, which each provide a specific functionality or logic related to blockchain networks.
The different Modules are the building blocks for the backend your blockchain application.
These building blocks can be assembled as desired to create a custom network, that is optimized to work seamlessly with your blockchain application.

1. Setup the Framework<br>
To get started in building your own blockchain application, set up the [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md).
The Modules come with a set of predefined default configuration, that let you start and test out your own network immediately.

2. Configure the Modules<br>
After the first successful start, feel free to configure, exchange or extend the Modules, suited to your specific needs.

3. Write own Modules<br>

> It is also possible to create own custom Modules from scratch, if needed.

Reviewer: Mention architecture and following the interfaces specs so the modules can communicate

For convenience, you may reuse the packages of [Lisk Elements](../lisk-sdk/lisk-elements/introduction.md) which provide related functionalities for creating blockchain applications, like cryptography.
Beside this, the **Lisk Framework** offers [Components](../lisk-sdk/lisk-framework/introduction.md).
Components consist of functionalities that are useful for most Modules, and provide an easy to use interface to perform actions on the application as a whole.
<br>The [Lisk Framework](../lisk-sdk/lisk-framework/introduction.md) initializes the Modules and controls their communication between each other and with the application as a whole.
To do this, each Module creates its' own communication channel, where it can publish events that it wants to share with the whole application.
Modules can passively listen to certain events that have been published by other Modules, or they can actively ask for information from other Modules.

4. Interact with your network<br>

> While your network is up and running, use [Lisk Commander](../lisk-sdk/lisk-commander/introduction.md) to interact with the network via the CLI (Command-Line-Interface).

Reviewer: Include a note of liskhq/cryptohraphy implemeting all of the needed crypto libraries

Author: Why here?

> 5. Build the frontend<br>

Reviewer: Remove that part for nm
```suggestion
```

When you have the network up and running as desired, it is time to build a nice frontend for your application, so that users are happy to use it.

### Interact with the network

> 

Reviewer: Explore the packages of Elements and describe Lisk Client.

**D. Lisk Hub**<br>
[Lisk Hub](https://lisk.io/hub) is the Graphical User Interface (GUI) for normal users to interact with the network.
It consists of normal wallet functionalities and more extended features, like delegate voting.

### Contribute to the project

First off, thanks for taking the time to contribute!

There are several ways how to participate in the Lisk project:

 How                       | Where
-------------------------  | -------------------------------------------------------------------------------------------------------------------
Contribute to the protocol | Discuss future changes of the protocol with the Science team on [research.lisk.io](https://research.lisk.io/).
Contribute to the codebase | [LiskHQ Github](https://github.com/LiskHQ) is the place to be! Be sure to follow our [Contribution Guidelines](https://github.com/LiskHQ/lisk-sdk/blob/development/docs/CONTRIBUTING.md) before submitting any code.
Give feedback              | For general feedback, please use [Lisk on Reddit](https://www.reddit.com/r/Lisk/) or [Lisk Chat](https://lisk.chat/), and [Gitter](https://gitter.im/LiskHQ/lisk) for technical questions and feedback.

> ## The Lisk Protocol

Reviewer: Before the "Contribute" section.

# Lisk Documentation

Welcome!

Welcome! For a smooth and efficient onboarding, we’ve updated our documentation to reflect the most common actions that developers take on our site so that you can easily find the information that is most relevant for you.

[What is Lisk?](https://lisk.io/documentation/lisk-protocol) Our vision at Lisk is to empower individuals to create a more decentralized, efficient and transparent global economy. We welcome you to join us in our mission.

## Getting started

For a quick & easy start, we provide getting started Guides for the most common use cases and actions people want to take:

Action                                                            | Description                                                                                                               
 ---------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

> [Build a blockchain application](build-blockchain-app.md)         | <PRE>1. Alpha SDK [step by step guide](build-blockchain-app.md#how-to-build-a-blockchain-application-with-the-lisk-sdk)<br>  a) How to [create](build-blockchain-app.md#1-setup-the-sdk) your custom blockchain<br>  b) How to [configure](build-blockchain-app.md#2-configure-and-customize-the-application) your blockchain<br>  c) How to create a [custom transaction](build-blockchain-app.md#2b-design-custom-transaction-types)<br>  d) Build the [frontend](build-blockchain-app.md#3-build-the-frontend) and [interact](interact-with-network.md) with the network<br>2. [Why blockchain?](https://lisk.io/academy/blockchain-basics/use-cases)</PRE>

Reviewer: Chapter names are not consistent with the file names, please do so and use the following titles:
1. How to set up and start Lisk SDK?
2. How to customise blockchain parameters to suit an application use-case?
3. How to add new custom transaction types to implement an application use-case?
4. How to interact with the network?

> [Maintain a node](maintain-node.md)                               | <PRE>1. [Deploy](maintain-node.md#how-to-set-up-a-node) your first blockchain node in 5 minutes<br>2. Maintain your existing \[[Binary](../lisk-core/administration/binary.md)&#124;[Docker](../lisk-core/administration/docker.md)&#124;[Source](../lisk-core/administration/source.md)\] node<br>3. Benefits of operating a node.</PRE>

Reviewer: Same as above - please unify the section file names and titles. Also, make the full sentences being the link here - "How to set up a node"

> [Interact with the network](interact-with-network.md)             | Interact with the blockchain network through the [API](https://lisk.io/documentation/lisk-core/api) and install useful [tools](interact-with-network.md#a-use-the-command-line) that facilitate the process.

Reviewer 2: ```suggestion
[Interact with the network](interact-with-network.md)             | Interact with the blockchain network through its [API](https://lisk.io/documentation/lisk-core/api) and install useful [tools](interact-with-network.md#a-use-the-command-line) to facilitate the process.
```

[Get to know the Lisk Protocol](../lisk-protocol/introduction.md) | Learn the underlying logic and technical concepts behind the Lisk Protocol including its [consensus algorithm](../lisk-protocol/consensus.md), [security](../lisk-protocol/security.md), [blocks](../lisk-protocol/blocks.md), [transactions](../lisk-protocol/transactions.md) and [peer communications](../lisk-protocol/p2p-communication.md).

## Lisk products

> For each product we provide a general introduction, step-by-step How-Tos and detailed specifications.

Reviewer 2: ```suggestion
For each product we provide a general introduction, step-by-step guides and detailed specifications.
```

Tool                                                         | Description
------------------------------------------------------------ | --------------------------------------------------

> [Lisk SDK](../lisk-sdk/introduction.md)                      |  Set of modular software products that provide developers with all the tools they need to develop their own custom blockchain applications.

Reviewer: In descriptions please reflect the GitHub READMEs: https://github.com/LiskHQ/lisk-sdk
